File: TypeScript/src/Entities/CatAdapter.ts
import { Cat } from "./Cat";
import { Dog } from "./Dog";
import { Entity } from "./Entity";

export class CatAdapter extends Cat {
    private readonly dog: Dog;

    constructor(dog: Dog) {
        super(dog.getName());
        this.dog = dog;
    }

    public meow(annoyedAt: Entity): string {
        return `${this.dog.bark()} at ${annoyedAt.getName()}`;
    }

    public getNoise(): string {
        return this.dog.getNoise();
    }

    public getName(): string {
        return this.dog.getName();
    }
}

File: Java/src/java/Entities/CatAdapter.java
import Cat;

public class CatAdapter extends Cat {
    private final Dog dog;

    public CatAdapter(Dog dog) {
        super(dog.getName());

        this.dog = dog;
    }

    @Override
    public String meow() {
        return dog.getNoise + " at nobody";
    }

    @Override
    String String getNoise() {
        dog.getNoise()
    }

    @Override
    void setNoise(String noise) {
        dog.setNoise(noise);
    }

    @Override
    String getName() {
        return dog.getName();
    }

}